rtt/base/TaskCore.cpp

```diff
@@ -337,5 +352,12 @@ namespace RTT {
         }
     }
 
+    std::string TaskCore::getName() const {
+        return name;
+    }
+    void TaskCore::setName(const std::string& name) {
+        this->name = name;
+        this->cName = strdup(name.c_str());
```

Comment:
Because c_str() is lifetime-sensitive, which does not play nice with the tracing. Since I needed a longer lifetime, I also preferred removing all issues related to memory allocation (since c_str() has no guarantee there either).